app/src/main/java/com/connorlay/knitgrid/activities/PatternDetailActivity.java

```diff
@@ -3,29 +3,55 @@
 import android.os.Bundle;
 import android.support.v4.app.FragmentTransaction;
 import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
 
 import com.connorlay.knitgrid.R;
+import com.connorlay.knitgrid.fragments.PatternGridLegendFragment;
 import com.connorlay.knitgrid.fragments.PatternHighlightFragment;
 import com.connorlay.knitgrid.models.Pattern;
 import com.connorlay.knitgrid.presenters.PatternPresenter;
 
 import butterknife.ButterKnife;
 
-public class PatternDetailActivity extends AppCompatActivity {
+public class PatternDetailActivity extends AppCompatActivity implements View.OnClickListener {
     public static final String STITCH_EXTRA = "STITCH_EXTRA";
     public static final String ARG_PATTERN = "PatternDetailActivity.ArgPattern";
+    private PatternPresenter patternPresenter;
+    private Pattern pattern;
```

Comment:
PatternGridLegendFragment.newInstance(pattern) uses the pattern which is why it is global.